### ISSUE TITLE
Possible Hang-Bug Found in Calendar_Generic_Page_Fragment

### ISSUE THREAD

**Issue by brok85** [ObjectiveTruth/UoitDCLibraryBooking#3]
Dear developers,
I am developing a tool to detect hang-bugs of android apps. A hang-bug happens when the execution of heavy APIs on the main thread lead the main thread to be unresponsive for an amount of time longer than the human-eye perceivable delay. We tested the master version of your app and found that the usage of the following Timber library API:

```
timber.log.Timber.v

timber.log.Timber.i
```

cause hang-bugs with the user perceiving a freeze of almost 1 second. We detect this hang-bug  specially when the app is opened but also in other interactions with the app. In particular, an extract of stack trace that calls the above API is the following:

```
1)com.objectivetruth.uoitlibrarybooking.Calendar_Generic_Page_Fragment.onCreateView(Calendar_Generic_Page_Fragment.java:114)

2)com.objectivetruth.uoitlibrarybooking.Calendar_Generic_Page_Fragment$CalendarAdapter.<init>(Calendar_Generic_Page_Fragment.java:293)

3)timber.log.Timber.v(Timber.java:13)
```

Moving heavy tasks off the main thread helps improving the responsiveness of the app. Therefore, we think that moving off the main thread (e.g., using AsyncTask for example) the execution of the Timber library APIs (Timber.i, Timber.v) in the CalendarAdapter method would fix this hang-bug and greatly help improving the responsiveness of your app.

Do you think that this approach would help improving the responsiveness of the app?

Thanks a lot for any reply,
Hope to help improving UoitDCLibraryBooking's responsiveness

**Comment by ObjectiveTruth** [comment]
Hey, thanks for the comment. 

You're right, there is an async issue but it should be taken care of by moving to RxJava #14 

:rocket:

**Comment by ObjectiveTruth** [comment]
Fixed in #14
